src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.cs

```diff
@@ -70,16 +87,46 @@ public static partial class JsonSerializer
                         }
                         else if (readStack.Current.IsProcessingDictionaryOrIDictionaryConstructible())
                         {
-                            HandleStartDictionary(options, ref readStack);
+                            readStack.EnqueueInitTask(InitTaskType.Dictionary);
```

Comment:
Please measure perf of this. This feature should not degrade perf when the feature is not active and this method will always create a dictionary.